packages/app/api-common/src/paginationUtils.ts

```diff
@@ -201,3 +201,22 @@ export async function getPaginatedEntriesByHasMore<T extends Record<string, unkn
 
   return resultArray
 }
+
+export async function traversePaginatedSourceByHasMore<
```

Comment:
As it stands, this is pretty much just a copy of `getPaginatedEntriesByHasMore`, with an addition of the processPage() call.

In the future, I can imagine something more parallelized, with apiCall being called in a promise pool as fast as the processPage() is able to process it (as I assume processPage will be the bottleneck), so that we wouldn't need to keep things in memory.